SConstruct

```diff
@@ -1226,6 +1226,9 @@ else:
 				"-MD",	# create multithreaded DLL
 				"-DBOOST_DISABLE_ASSERTS",
 				"-Ox",
+				# -Og optimization (included via -Ox) generates lots of unreachable 
+				# code warnings from boost::intrusive_ptr. Disabled in release build only.
+				"/wd4702"
```

Comment:
Out of interest, do you know why `external:I` isn't suppressing these anyway?

Reply:
I'm not sure about that one, but one guess is that an optimization is inlining from `boost::intrusive_ptr` which is dragging the warning into our code? The warning is a result of the declaration of [boost::intrusive_ptr](https://github.com/boostorg/smart_ptr/blob/e37cd4154f492b3cd2ea8e87806614ffddf1163a/include/boost/smart_ptr/intrusive_ptr.hpp#L63). I found a couple of similar sounding issues in searching, but nothing that seems like a direct match.

https://stackoverflow.com/questions/15069526/unreachable-code-warning-in-simple-single-line-function
https://github.com/boostorg/beast/issues/1582

The second link suggests it may be fixed when we move to Visual Studio 2019.